examples/chef/chef.py

```diff
@@ -490,9 +492,17 @@ def main(argv: Sequence[str]) -> None:
     #
 
     if options.do_build:
-        branch = shell.run_cmd(
-            "git branch | awk -v FS=' ' '/\*/{print $NF}' | sed 's|[()]||g'", return_cmd_output=True).replace("\n", "")
-        commit_id = shell.run_cmd("git rev-parse HEAD", return_cmd_output=True).replace("\n", "")
+        if options.do_automated_test_stamp:
+            branch = shell.run_cmd(
+                "git branch | awk -v FS=' ' '/\*/{print $NF}' | sed 's|[()]||g'", return_cmd_output=True).replace("\n", "")
+            commit_id = shell.run_cmd("git rev-parse HEAD", return_cmd_output=True).replace("\n", "")
+            sw_ver_string = f"""{branch}:{commit_id}"""
```

Comment:
There are many cases where branch name isn't relevant, or perhaps doesn't exist at all.  Why not consider git describe?

Reply:
Yes, you are right,  some time the branch doesn't exist at all so we couldn't get the branch name by `$git branch --show-current` simply.
That's also why I do some parsing work by sed/awk to get some branch info from `$git branch`.
And I think the `git describe --tag` you mentioned also a good idea. I'll discuss with teammate about this.